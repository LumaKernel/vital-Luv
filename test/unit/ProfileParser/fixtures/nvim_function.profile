SCRIPT  /home/luma/vim-make-plugin/vital-Luv/dev/ProfileParser/test_function.vim
Sourced 1 time
Total time:   0.000153
 Self time:   0.000107

count  total (s)   self (s)
                            
    1              0.000005 function! F()
                              echomsg 'string'
                            endfunction
                            
    1   0.000056   0.000010 call F()
                            

FUNCTION  F()
    Defined: ~/vim-make-plugin/vital-Luv/dev/ProfileParser/test_function.vim line 2
Called 1 time
Total time:   0.000046
 Self time:   0.000046

count  total (s)   self (s)
    1              0.000025   echomsg 'string'

FUNCTIONS SORTED ON TOTAL TIME
count  total (s)   self (s)  function
    1   0.000046             F()

FUNCTIONS SORTED ON SELF TIME
count  total (s)   self (s)  function
    1              0.000046  F()

