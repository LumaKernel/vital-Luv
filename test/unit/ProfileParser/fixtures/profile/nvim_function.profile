SCRIPT  /home/luma/vim-make-plugin/vital-Luv/dev/ProfileParser/test_function.vim
Sourced 1 time
Total time:   0.000096
 Self time:   0.000071

count  total (s)   self (s)
                            
    1              0.000003 function! F()
                              echomsg 'string'
                            endfunction
                            
    1   0.000057   0.000031 call F()
                            

FUNCTION  F()
    Defined: ~/vim-make-plugin/vital-Luv/dev/ProfileParser/test_function.vim line 2
Called 1 time
Total time:   0.000025
 Self time:   0.000025

count  total (s)   self (s)
    1              0.000018   echomsg 'string'

FUNCTIONS SORTED ON TOTAL TIME
count  total (s)   self (s)  function
    1   0.000025             F()

FUNCTIONS SORTED ON SELF TIME
count  total (s)   self (s)  function
    1              0.000025  F()

