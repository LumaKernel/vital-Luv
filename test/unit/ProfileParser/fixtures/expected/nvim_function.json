[{"path":"/test/test_function.vim","total_time":1.03e-4,"self_time":7.9e-5,"type":"script","lines":[{"content":""},{"self_time":5.0e-6,"count":1,"content":"function!"},{"content":"      \\ F()"},{"content":"  echomsg"},{"content":"   \\ 'string'"},{"content":"   \\ . 'string'"},{"content":"  echo \"hello\""},{"content":"endfunction"},{"content":""},{"total_time":4.3e-5,"self_time":1.9e-5,"count":1,"content":"call F()"},{"content":""}],"count":1},{"defined":{"path":"/test/test_function.vim","line":3},"name":"F()","total_time":2.3e-5,"self_time":2.3e-5,"type":"function","lines":[{"self_time":1.4e-5,"count":1,"content":"  echomsg 'string' . 'string'"},{"self_time":3.0e-6,"count":1,"content":"  echo \"hello\""}],"count":1},{"type":"function_list","what":"TOTAL TIME","functions":[{"name":"F()","total_time":2.3e-5,"count":1}]},{"type":"function_list","what":"SELF TIME","functions":[{"name":"F()","self_time":2.3e-5,"count":1}]}]
